### PR TITLE
Prep for 4.2.3 release

### PR DESCRIPTION
**Pull request by MGatner**
**Description**
Updates changelog and version references for `4.2.3`.

**Checklist:**
- [X] Securely signed commits
- [X] Component(s) with PHPDoc blocks, only if necessary or adds value
- [X] Unit testing, with >80% coverage
- [X] User guide updated
- [X] Conforms to style guide

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,5 +1,10 @@
 # Changelog
 
+## [v4.2.3](https://github.com/codeigniter4/CodeIgniter4/tree/v4.2.3) (2022-08-06)
+[Full Changelog](https://github.com/codeigniter4/CodeIgniter4/compare/v4.2.2...v4.2.3)
+
+* security: Improve CSRF protection (for Shield CSRF security fix)
```
Comment (kenjis) [inline CHANGELOG.md:6]:
```suggestion
* SECURITY: Improve CSRF protection (for Shield CSRF security fix)
```

**Review by kenjis — APPROVED** [review]

**Comment by kenjis** [comment]
We should have checked the warning for the docs:
https://github.com/codeigniter4/CodeIgniter4/pull/6353/files#diff-f9970d86aca13e5c6ae72e6f0a0b8aaac1f242741a364ce3a960bc3866496ccb
> Check warning on line 1 in user_guide_src/source/installation/upgrade_423.rst
GitHub Actions / Check User Guide syntax
user_guide_src/source/installation/upgrade_423.rst#L1
document isn't included in any toctree

**Comment by MGatner** [comment]
That didn't show up anywhere for me. Is there any way that can elevate to an actual error so the Action will fail? @paulbalandan might know?

**Comment by kenjis** [comment]
This worked!

```console
$ make html SPHINXOPTS="-W"
Running Sphinx v4.5.0
making output directory... done
building [mo]: targets for 0 po files that are out of date
building [html]: targets for 207 source files that are out of date
updating environment: [new config] 207 added, 0 changed, 0 removed
reading sources... [100%] tutorial/static_pages                                                                                                                      
looking for now-outdated files... none found
pickling environment... done
checking consistency... done
preparing documents... done
writing output... [100%] tutorial/static_pages                                                                                                                       

Warning, treated as error:
/Users/kenji/work/codeigniter/CodeIgniter4/user_guide_src/source/database/utilities.rst:20:Could not lex literal_block as "html+php". Highlighting skipped.
make: *** [html] Error 2
```
https://stackoverflow.com/questions/38048945/how-to-turn-warnings-into-errors-when-building-sphinx-documentation-with-setupto

**Comment by kenjis** [comment]
I sent a PR #6359

**Pull request merged**